### ISSUE TITLE
transmitting dotenv env variables from .env.development, .env.staging, .env.production when precompiling assets locally

### ISSUE THREAD

**Issue by Leooo** [thoughtbot/ember-cli-rails#208]
Hello, I spent many hours on this and haven't be able to isolate the problem, so I thought maybe you could have an idea:

I'm using the 'dotenv-rails' gem to make my .env variables accessible to Ember. Following https://github.com/rwz/ember-cli-rails/issues/19, I was able to make the env variables accessible via using process.env in config/environment.js:
`````
//in config.environment.js
  var ENV = {
    dotenv: process.env,
   ..
  }
`````

I'm using different files for the env variables: one .env file shared by all environments, and .env.development, .env.staging, .env.production for each corresponding environment.

My problem is that when running the app on Heroku in staging environment, dotenv variables are still the ones of .env/.env.development files, (not the ones of .env.staging as would be expected) and I don't understand why?

Some precisions:
* I am not using npm dotenv package (as you precised in https://github.com/rwz/ember-cli-rails/issues/19 there is no need for it).
* the 'dotenv-rails' gem is in the top of my gemfile, before 'ember-cli-rails'
* I'm using heroku deploy, with a local precompilation of assets before deploying (in order to populate cloudfront before the deployment phase). So a variation of rake assets:precompile RAILS_ENV=staging
* opening the rails console for the staging application, I do see the correct env variables as set by .env.staging.  So the problem is really on Ember's process.env side.

Any Idea? Looking at your code I have no idea how the transmission of the ENV variables to Ember works. Many thanks

**Comment by Leooo** [comment]
Alternatively, is it possible to pass the env variables of heroku config:set to the ember app? I see this is mentioned in https://github.com/ember-cli/ember-cli/issues/1236, but possibly linked to the heroku-builpack package that we are not using for ember-cli-rails ?!

**Comment by Leooo** [comment]
**tl;dr**:
It would still be useful to have Ember-cli-rails recognize the correct environment variables on .env.staging when pre-compiling locally with RAILS_ENV=staging.

**Long story**:
The .env.staging variables are understood by Ember when Heroku precompiles the assets during the deployment (had to do a rake assets:clobber and set NODE_MODULES_CACHE=false to make it work, in case someone else is trying to do that). The problem now is that I'm using a CDN to fetch my assets, so /app/assets is in .slugignore to minimize the slug size. But now that /app/assets are not in produciton I can not use anymore the trick of @gniquil in https://github.com/rwz/ember-cli-rails/pull/119  to get my asset paths in Ember:
`````
  window.assets.images = <%= raw(image_assets.to_json) %>;
`````
, which was working when I was pre-compiling my assets locally.

**Comment by seanpdoyle** [comment]
@Leooo  Hey there, sorry this has gone silent.

There's been a lot of movement on this repo recently.

Could you try this again for version [0.5.3](https://rubygems.org/gems/ember-cli-rails/versions/0.5.3)?

**Comment by seanpdoyle** [comment]
Could you try the [latest version](https://github.com/thoughtbot/ember-cli-rails/releases/latest)?

I'm closing this issue due to it's age. If this is still a problem, say so and we can take another shot.
